geode-core/src/main/java/org/apache/geode/internal/cache/PRHARedundancyProvider.java

```diff
@@ -1774,7 +1775,9 @@ public void run2() {
       }
     } finally {
       for (final ProxyBucketRegion proxyBucket : bucketsNotHostedLocally) {
-        allBucketsRecoveredFromDisk.countDown();
+        if (getPersistentBucketRecoverer() != null) {
+          getPersistentBucketRecoverer().countDown();
```

Comment:
have a flavor of countDown(int) and call countDown(bucketsNotHostedLocally.size())